modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/BulkCopyEntriesMVCActionCommand.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.document.library.web.internal.portlet.action;
+
+import com.liferay.depot.group.provider.SiteConnectedGroupGroupProvider;
+import com.liferay.document.library.constants.DLPortletKeys;
+import com.liferay.document.library.kernel.model.DLFileEntry;
+import com.liferay.document.library.kernel.model.DLFileShortcut;
+import com.liferay.document.library.kernel.model.DLFolder;
+import com.liferay.document.library.kernel.service.DLAppService;
+import com.liferay.document.library.kernel.service.DLFileEntryLocalService;
+import com.liferay.document.library.kernel.service.DLFileShortcutLocalService;
+import com.liferay.document.library.kernel.service.DLFolderLocalService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceContextFactory;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Sam Ziemer
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + DLPortletKeys.DOCUMENT_LIBRARY,
+		"javax.portlet.name=" + DLPortletKeys.DOCUMENT_LIBRARY_ADMIN,
+		"javax.portlet.name=" + DLPortletKeys.MEDIA_GALLERY_DISPLAY,
+		"mvc.command.name=/document_library/bulk_copy_entries"
+	},
+	service = MVCActionCommand.class
+)
+public class BulkCopyEntriesMVCActionCommand extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		long destinationFolderId = ParamUtil.getLong(
+			actionRequest, "destinationFolderId");
+		long destinationRepositoryId = ParamUtil.getLong(
+			actionRequest, "destinationRepositoryId");
+		long[] entityIds = ParamUtil.getLongValues(
+			actionRequest, "selectedEntries");
```

Comment:
so for this you should do several things:

1st:
on `bulk_copy_entries.jsp`
do something like
```
).put("selectedEntries", dlCopyEntryDisplayContext.getSelectedEntries()
```

adding the new method on 
`DLCopyEntryDisplayContext`
something like:
```
public long[] getSelectedEntries() {
  if (ArrayUtil.isNotEmpty(_selectedEntries)) {
	  return _selectedEntries;
  }
  
  _selectedEntries = ParamUtil.getLongValues(
	  _httpServletRequest, "selectedEntries");
  
  return _selectedEntries;
  }
```
This will pass the new parameter to the DDMFolderSelector.js


```
const DDMFolderSelector = ({
	copyActionURL,
	fileShortcutId,
	itemType,
	portletNamespace,
	redirect,
	selectedEntries,
	selectionModalURL,
	sourceFileEntryId,
	sourceFileName,
	sourceFolderId,
	sourceFolderName,
	sourceRepositoryId,
}) => {
```

and the pass to the submit

```
: {
    [`${portletNamespace}selectedEntries`]: selectedEntries,
    [`${portletNamespace}fileEntryId`]: sourceFileEntryId,
    [`${portletNamespace}fileShortcutId`]: fileShortcutId,
    [`${portletNamespace}destinationFolderId`]: destinationParentFolderId,
    [`${portletNamespace}destinationRepositoryId`]: destinationRepositoryId,
}
```

This way the on the action command

`long[] entityIds = ParamUtil.getLongValues(actionRequest, "selectedEntries");`
Should have values

Reply:
maybe the part of the can be more accurate than my approach.
@ambrinchaudhary  maybe can help you with the react part